modules/auxiliary/admin/kerberos/pkinit_login.rb

```diff
@@ -65,12 +65,13 @@ def run
     print_status("Attempting PKINIT login for #{username}@#{realm}")
     begin
       server_name = "krbtgt/#{realm}"
-      tgt_result, key = send_request_tgt_pkinit(pfx: pfx,
-                                                username: username,
-                                                realm: realm,
-                                                server_name: server_name)
+      tgt_result = send_request_tgt_pkinit(
+        pfx: pfx,
+        username: username,
+        realm: realm,
+        server_name: server_name
+      )
       print_good('Successfully authenticated with certificate')
-      enc_part = decrypt_kdc_as_rep_enc_part(tgt_result.as_rep, key)
 
       info = []
       info << "realm: #{realm.upcase}"
```

Comment:
This needs to use a different pattern - a similar solution will be required to make https://github.com/rapid7/metasploit-framework/pull/17226 have less copy/paste

It makes to me sense to not block on that pattern for this PR to land